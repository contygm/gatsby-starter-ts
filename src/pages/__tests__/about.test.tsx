import React from 'react';
import { render } from '@testing-library/react';
import About, { Head } from '../about';

describe('About', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<About />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('header meta data renders correctly', () => {
        const { asFragment } = render(<Head />);
        expect(asFragment()).toMatchSnapshot();
    });
});
