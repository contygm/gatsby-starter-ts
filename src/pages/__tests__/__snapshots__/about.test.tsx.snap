// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About header meta data renders correctly 1`] = `
<DocumentFragment>
  <title>
    About
  </title>
  <meta
    content="a super duper site!"
    name="description"
  />
  <meta
    content="https://www.domain.com/nothing.png"
    name="image"
  />
  <link
    href="../../images/icon.png"
    rel="icon"
  />
</DocumentFragment>
`;

exports[`About renders correctly 1`] = `
<DocumentFragment>
  <div
    id="app-container"
  >
    <nav
      aria-label="main navigation"
      class="navbar is-primary"
      role="navigation"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
          href="/"
        >
          <img
            alt="Logo"
            data-cy="nav-link-logo"
            layout="constrained"
            placeholder="blurred"
            src="../../images/bulma-logo-white.png"
            width="150"
          />
        </a>
        <button
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger "
          data-cy="nav-hamburger"
          data-testid="siteNav"
        >
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
        </button>
      </div>
      <div
        class="navbar-menu "
        id="siteNav"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            data-cy="nav-link-home"
            href="/"
          >
            Home
          </a>
          <a
            class="navbar-item"
            data-cy="nav-link-blog"
            href="/blog"
          >
            Blog
          </a>
          <a
            class="navbar-item"
            data-cy="nav-link-wiki"
            href="/wiki"
          >
            Wiki
          </a>
          <a
            class="navbar-item"
            data-cy="nav-link-glossary"
            href="/glossary"
          >
            Glossary
          </a>
        </div>
      </div>
    </nav>
    <main>
      <header
        class="medium-hero"
      >
        <div
          class="pg-header-hero-black"
        >
          <div
            class="container "
          >
            <h1
              class="pg-header-title"
            >
              About The Company
              <span
                class="pg-header-period"
              >
                .
              </span>
            </h1>
          </div>
        </div>
      </header>
      <article
        class="content"
      >
        <section
          class="section"
        >
          <div
            class="container"
          >
            <h2
              class="title-two"
            >
              Our Story
            </h2>
            <p>
              Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé. Muffin cake halvah marshmallow carrot cake powder. Lemon drops muffin biscuit marzipan jujubes cupcake.
            </p>
            <p>
              Biscuit cake candy brownie sweet bear claw croissant pastry jelly-o. Dessert pie marzipan liquorice apple pie toffee shortbread pie sweet roll. Candy canes halvah icing dragée toffee sweet dessert topping chocolate bar.
            </p>
            <p>
              Icing marzipan sugar plum danish chocolate candy canes gummies. Chocolate cotton candy candy halvah jujubes toffee jelly beans ice cream. Sweet roll candy tart gummi bears danish icing biscuit apple pie. Chocolate cake gummies sugar plum candy canes pudding macaroon.
            </p>
          </div>
        </section>
        <section
          class="section"
        >
          <div
            class="container"
          >
            <h2
              class="title-two"
            >
              Our Tech
            </h2>
            <p>
              Bonbon tootsie roll lemon drops topping biscuit cookie. Icing gummi bears ice cream marshmallow jelly brownie icing soufflé. Muffin cake halvah marshmallow carrot cake powder. Lemon drops muffin biscuit marzipan jujubes cupcake.
            </p>
            <p>
              Biscuit cake candy brownie sweet bear claw croissant pastry jelly-o. Dessert pie marzipan liquorice apple pie toffee shortbread pie sweet roll. Candy canes halvah icing dragée toffee sweet dessert topping chocolate bar.
            </p>
            <ul>
              <li>
                 GatsbyJS 
              </li>
              <li>
                 Typescript 
              </li>
              <li>
                 NetlifyCMS 
              </li>
              <li>
                 Bulma 
              </li>
            </ul>
            <p>
              Icing marzipan sugar plum danish chocolate candy canes gummies. Chocolate cotton candy candy halvah jujubes toffee jelly beans ice cream. Sweet roll candy tart gummi bears danish icing biscuit apple pie. Chocolate cake gummies sugar plum candy canes pudding macaroon.
            </p>
          </div>
        </section>
        <section
          class="section"
        >
          <div
            class="container"
          >
            <h2
              class="title-two"
              id="contact-us"
            >
              Contact Us
            </h2>
            <p>
              Have any questions, comments or concerns for The Company? We would love to hear from you. Feel free to reach out to us using our contact information below:
            </p>
            <ul>
              <li>
                 Email: email@email.com
              </li>
              <li>
                 Phone Number: 999-999-9999 
              </li>
            </ul>
            <p>
              Biscuit cake candy brownie sweet bear claw croissant pastry jelly-o. Dessert pie marzipan liquorice apple pie toffee shortbread pie sweet roll. Candy canes halvah icing dragée toffee sweet dessert topping chocolate bar.
            </p>
            <div
              class="contact-form-container"
            >
              <div
                class="contact-form-wrapper"
              >
                <form
                  class="is-small"
                  enctype="text/plain"
                  method="get"
                >
                  <div
                    class="field"
                  >
                    <label
                      class="label"
                    >
                      Name
                    </label>
                    <div
                      class="control"
                    >
                      <input
                        class="input"
                        name="name"
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="field"
                  >
                    <label
                      class="label"
                    >
                      YourEmail
                    </label>
                    <div
                      class="control"
                    >
                      <input
                        class="input"
                        name="email"
                        placeholder="email@email.com"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="field"
                  >
                    <label
                      class="label"
                    >
                      Message
                    </label>
                    <div
                      class="control"
                    >
                      <textarea
                        class="textarea"
                        name="message"
                        placeholder="Your messages"
                      />
                    </div>
                  </div>
                  <div
                    class="contact-action-buttons"
                  >
                    <div
                      class="control"
                    >
                      <button
                        class="primary-button"
                        type="submit"
                      >
                        Submit
                      </button>
                    </div>
                    <div
                      class="control"
                    >
                      <button
                        class="light-grey-button"
                        name="reset"
                        type="reset"
                      >
                        Cancel
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </section>
      </article>
    </main>
    <footer
      class="footer"
    >
      <div
        class="footer-main-content-wrapper"
      >
        <div
          class="column"
        >
          <a
            class=""
            data-cy="footer-logo"
            href="/"
          >
            <figure
              class="footer-logo"
            >
              <img
                alt="Logo"
                layout="constrained"
                placeholder="blurred"
                src="../../images/bulma-logo-white.png"
                width="128"
              />
            </figure>
          </a>
        </div>
        <ul
          class="centered-col"
          role="list"
          style="flex-basis: auto;"
        >
          <div>
            <li
              class="footer-pg-link-wrapper"
              role="listitem"
            >
              <a
                class="footer-link-base"
                data-cy="footer-link-Contact"
                href="/about#contact-us"
              >
                Contact
              </a>
            </li>
            <li
              class="footer-pg-link-wrapper"
              role="listitem"
            >
              <a
                class="footer-link-base"
                data-cy="footer-link-About"
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="footer-pg-link-wrapper"
              role="listitem"
            >
              <a
                class="footer-link-base"
                data-cy="footer-link-Privacy"
                href="/legal/privacy-policy"
              >
                Privacy
              </a>
            </li>
            <li
              class="footer-dropup"
              role="listitem"
            >
              <div
                class="dropdown-trigger"
              >
                <a
                  aria-controls="footer-dropdown-menu"
                  aria-haspopup="true"
                  class="footer-dropup-menu"
                  data-cy="footer-link-legal"
                  href="/legal"
                >
                  Legal
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-angle-up fa-sm px-2"
                    data-icon="angle-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="dropdown-menu"
                id="footer-dropdown-menu"
                role="menu"
              >
                <div
                  class="dropdown-content"
                >
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-copyright"
                      href="/legal/copyright"
                    >
                      Copyright
                    </a>
                  </div>
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-cookie-policy"
                      href="/legal/cookie-policy"
                    >
                      Cookie Policy
                    </a>
                  </div>
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-terms-and-conditions"
                      href="/legal/terms-and-conditions"
                    >
                      Terms and Conditions
                    </a>
                  </div>
                  <div
                    class="dropdown-item"
                  >
                    <a
                      class="footer-drawer-link"
                      data-cy="footer-link-disclaimer"
                      href="/legal/disclaimer"
                    >
                      Disclaimer
                    </a>
                  </div>
                </div>
              </div>
            </li>
          </div>
        </ul>
        <div
          class="column"
        >
          <ul
            class="footer-social-media"
          >
            <li
              class="faicon-link-container"
            >
              <a
                aria-label="social media link"
                class="faicon-link  is-white"
                data-cy="faicon-link"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-star fa-xl icon"
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
            <li
              class="faicon-link-container"
            >
              <a
                aria-label="social media link"
                class="faicon-link  is-white"
                data-cy="faicon-link"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-mug-hot fa-xl icon"
                  data-icon="mug-hot"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
            <li
              class="faicon-link-container"
            >
              <a
                aria-label="social media link"
                class="faicon-link  is-white"
                data-cy="faicon-link"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-ghost fa-xl icon"
                  data-icon="ghost"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 384 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="footer-copyright"
      >
        <p
          class="pt-3"
          data-testid="copyright"
        >
          © 
          <time>
            2023
          </time>
          , Built by Writer B. Author
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
